keps/sig-node/3008-qos-class-resources/README.md

```diff
@@ -0,0 +1,2204 @@
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Add support to Kubernetes for declaring _quality-of-service_ resources, and
+assigning these to Pods. A quality-of-service (QoS-class) resource is similar
+to other Kubernetes resource types (i.e. native resources such as `cpu` and
+`memory` or extended resources) because you can assign that resource to a
+particular container. However, QoS-class resources are also different from
+those other resources because they are used to assign a _class identifier_,
+rather than to declare a specific amount of capacity that is allocated.
+
+Main characteristics of the new resource type (and the technologies they are
+aimed at enabling) are:
+
+- multiple containers can be assigned to the same class of a certain type of
+  QoS-class resource
+- QoS-class resources are represented by an enumerable set of class identifiers
+- each type of QoS-class resource has an independent set of class identifiers
+
+With QoS-class resources, Pods and their containers can request opaque
+QoS-class identifiers (classes) of certain QoS mechanism (QoS-class resource
+type). Kubelet relays this information to the container runtime, without
+directing how the request is enforced in the underlying system. Being opaque to
+Kubernetes means that QoS-class resources have to be supported by the container
+runtime as it is responsible for the actual low-level management of them.
+
+## Motivation
+
+This enhancement proposal aims at improving the quality of service of
+applications by making available a new type of resource control mechanism in
+Kubernetes. Certain types of resources are inherently shared by applications,
+e.g. cache, memory bandwidth and disk I/O. While there are technologies for
+controlling how these resources are shared between applications, there is
+currently no meaningful way to support these technologies in Kubernetes. This
+proposal suggests to address the issue above in a generalized way by extending
+the Kubernetes resource model with a new type of resources, i.e. QoS-class
+resources.
+
+This KEP identifies two technologies that can immediately be enabled with
+QoS-class resources. However, these are just two examples  and the proposed
+changes are generic (and not tied to these two QoS-class resource types in any
+way), making it easier to implement new QoS-class resource types.
+
+[Intel RDT][intel-rdt] implements a class-based mechanism for controlling the
+cache and memory bandwidth QoS of applications. All processes in the same
+hardware class share a portion of cache lines and memory bandwidth. RDT
+provides a way for mitigating noisy neighbors and fulfilling SLAs. In Linux
+control happens via resctrl -- a pseudo-filesystem provided by the kernel which
+makes it virtually agnostic of the hardware architecture. The OCI runtime-spec
+has supported Intel RDT for a while already. Other hardware vendors have
```

Comment:
> The OCI runtime-spec has supported Intel RDT for a while already.

Is this true?

Reply:
Yes: https://github.com/opencontainers/runtime-spec/blob/main/config-linux.md#intelrdt

Reply:
Maybe link to https://github.com/opencontainers/runtime-spec/blob/v1.0.2/config-linux.md#IntelRdt

Reply:
Link added

Reply:
i.e. fixed